grading/unitex-nlp/Dockerfile

```diff
@@ -0,0 +1,32 @@
+# DOCKER-VERSION 1.1.0
+
+# This file was written by Anaïs Tack <anais.tack@uclouvain.be> for the course LFIAL2620 on INGInious.
+# This Dockerfile creates an INGInious grading environment with the Unitex/GramLab NLP engine.
+
+# By default, the latest version of INGInious will be used.
+# You can install a different version with --build-arg VERSION=<version>.
+ARG     VERSION=latest
+
+# Instead of using ingi/inginious-c-base (default configuration),
+# we must use the container with C/C++ (GCC) to install the Unitex/GramLab C++ Core.
+FROM    ingi/inginious-c-cpp:${VERSION}
+LABEL   org.inginious.grading.name="unitex-nlp"
+
+# By default, Unitex v3.3 will be installed. 
+# You can install a different version with --build-arg UNITEX_VERSION=<version>.
+ARG     UNITEX_VERSION=3.3
+
+# Install Unitex/GramLab C++ Core.
+WORKDIR /home
+ADD     https://github.com/UnitexGramLab/unitex-core/archive/refs/tags/v${UNITEX_VERSION}.tar.gz ./
+RUN     tar -xzf v${UNITEX_VERSION}.tar.gz --directory ./ && rm v${UNITEX_VERSION}.tar.gz
```

Comment:
Running `rm` here will probably not be as profitable as expected. Each Docker command results in a new container layer. To save space, you can issue a single `RUN` starting with a `wget` command to download the file. Running a `rm` command in another layer will use twice the space (both a positive and negative diff).

Reply:
Thanks! That's good to know. I tried using `wget` before but it seems it's unavailable in the base container and needs to be installed with `yum install -y wget`. Is that correct? So, I'll edit it into one run: `yum install -y wget && ... && rm v${UNITEX_VERSION}.tar.gz`.

Reply:
Indeed, it's correct.